demoIsoObject.cpp

```diff
@@ -147,6 +147,14 @@ class myObject : public ISO22133::TestObject {
         if (this->getObjectSettings().testMode == TEST_MODE_ONLINE){
             std::cout << "Test mode is online planned, appending new trajectory to existing" << std::endl;
             this->trajectory.insert(this->trajectory.end(), newTraj.begin(), newTraj.end());
+
+            // We might receive trajectories that overlap, we remove the duplicate points by checking the time
+            std::sort(this->trajectory.begin(), this->trajectory.end(), [](const TrajectoryWaypointType& t1, const TrajectoryWaypointType& t2) {
+                return t1.relativeTime.tv_sec * 1000000 + t1.relativeTime.tv_usec < t2.relativeTime.tv_sec * 1000000 + t2.relativeTime.tv_usec;
+            });
+            this->trajectory.erase(std::unique(this->trajectory.begin(), this->trajectory.end(), [](const TrajectoryWaypointType& t1, const TrajectoryWaypointType& t2) {
+                return t1.relativeTime.tv_sec * 1000000 + t1.relativeTime.tv_usec == t2.relativeTime.tv_sec * 1000000 + t2.relativeTime.tv_usec;
+            }), this->trajectory.end());
```

Comment:
There is an assumption here that overlapping times are exact. Maybe this check could be made on positions somehow instead? But we do not know how the trajectories will be produced yet so maybe this is a good first assumption..

Reply:
Yes, it could probably be improved in the future. I think some mix of time and positions might be used, because if we only check positions there could probably be problems if the trajectory overlaps itself (e.g. the object is running in a circle). I think this is good enough for now, and if we see a use case where we'll need to improve it we'll do it then.